source/INI/PeanutButter.INI.Tests/TestINI.cs

```diff
@@ -790,6 +790,8 @@ public void Persist_GivenStream_ShouldWriteOutToStream()
                 sut.AddSection("general");
                 sut["general"]["foo"] = "bar";
 
+                sut.AddSection("general2");
+                sut["general2"]["foo2"] = "bar2";
```

Comment:
the test adds a new section, but doesn't assert anything about that section -- I'd prefer a new test which does:
1. add two sections
2. persist (can persist to memstream & read out string)
3. assert that the section gap has been added